Паутина

KloopRE opened the pull request.

KloopRE committed:
lab2

KloopRE committed:
version 1.01

hBuzzy commented on widget.h:
```diff
@@ -2,12 +2,34 @@
+  void setCursorStyle();  // Добавим полное объявление метода setCursorStyle
+
+ private:
+  bool drawing_;
```
Переменные и поля типа `bool` должны задавать вопрос, на который мы можем строго ответить либо "Да", либо "Нет". Достигается это путем использования глаголов is, have, has и т.д. "Is drawing?" -> isDrawing_;

Соответственно в конструкции 
```C++
if (isDrawing_) {
}
```

мы можем точно понять когда у нас `true`, когда `false`

hBuzzy commented on widget.cpp:
```diff
@@ -1,5 +1,99 @@
+  painter->setPen(QPen(lineColor_, lineWidth_, Qt::SolidLine));
+
+  // Нарисуйте линии, представляющие паутину
+  for (int i = 0; i < 8; ++i) {
+    double angle = i * 45 * M_PI / 180.0;
+    int x = center.x() + 50 * qCos(angle);
+    int y = center.y() + 50 * qSin(angle);
+    painter->drawLine(center, QPoint(x, y));
+  }
```
8 - магическое число, его точно придется выделить в переменную с понятным названием.

Остальные переменные, в целом, тоже бы неплохо вынести.

hBuzzy commented on widget.cpp:
```diff
@@ -1,5 +1,99 @@
+  }
+
+  // Соединим линии друг с другом для формирования паутины
+  for (int i = 0; i < 8; ++i) {
+    double angle1 = i * 45 * M_PI / 180.0;
+    double angle2 = (i + 1) * 45 * M_PI / 180.0;
+
+    int x1 = center.x() + 50 * qCos(angle1);
+    int y1 = center.y() + 50 * qSin(angle1);
+
+    int x2 = center.x() + 50 * qCos(angle2);
+    int y2 = center.y() + 50 * qSin(angle2);
+
+    painter->drawLine(QPoint(x1, y1), QPoint(x2, y2));
+  }
+}
```
Аналогично - магические числа.

hBuzzy reviewed (changes requested):
Исправить замечания. Всю логику отрисовки выделить в класс "Паутина".

В виджете должен быть переопределен метод paintEvent (он сейчас есть), где у объекта класса Паутина вызывается метод Web::draw(QPainter &painter);

KloopRE committed:
Исправление замечаний.

В виджете должен быть переопределен метод paintEvent (он сейчас есть), где у объекта класса Паутина вызывается метод Web::draw(QPainter &painter);

Переменные и поля типа bool должны задавать вопрос, на который мы можем строго ответить либо "Да", либо "Нет". Достигается это путем использования глаголов is, have, has и т.д. "Is drawing?" -> isDrawing_;

Убраны магические числа

hBuzzy commented on web.cpp:
```diff
@@ -1,3 +1,42 @@
+}
+
+void Web::drawLines(QPainter *painter, const QPoint &center) {
+    const int numLines = 8;
```
Не сокращайте названия. Num, конечно, +- устоявшееся сокращение, но все же лучше numberOfLines или еще лучше linesNumber.

hBuzzy commented on widget.h:
```diff
@@ -24,10 +25,11 @@ class Widget : public QWidget {
 
  private:
-  bool drawing_;
+  bool isdrawing_;
```
isdrawing_ -> isDrawing_

Используем camelCase в именовании полей. Первое слово с маленькой буквы, все остальные - с большой.

hBuzzy reviewed (commented):
Принято.

Обратите внимание на замечания.
